smartlearner/tasks.py

```diff
@@ -23,6 +25,15 @@ def init(self, status):
     def finished(self, status):
         print("Training done in {:.03f} sec.".format(time() - self.start_time))
 
+    def save(self, path):
+        state = {"version": 1,
+                 "start_time": self.start_time}
+        utils.save_dict_to_json_file(pjoin(path, "print_training_duration.json"), state)
```

Comment:
That I don't really care for human readability, so long as the titles of the files gives enough info for you to know what's actually inside. If you're scared to forget what was your hyperparameters, you could always code a small exploration function that would load a file in a dict to be able to have a look at it inside ipython.

Conclusion: choose one, I don't really care which one.